src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -2372,11 +2376,8 @@ private boolean isFeedbackSessionViewableTo(
         }
 
         // Allow all instructors to view always
-        InstructorAttributes instructor = instructorsLogic.
-                getInstructorForEmail(session.courseId, userEmail);
-        if (instructor != null) {
-            return instructorsLogic.isEmailOfInstructorOfCourse(
-                    instructor.email, session.courseId);
+        if (instructorOfCourse != null) {
+            return true;
         }
```

Comment:
So this is where the instructor object is being checked for null. That's not the right place. It makes the method less cohesive and unnecessarily coupled with the caller method above (i.e. one logical action split into two methods).